docs/routing.md

```diff
@@ -0,0 +1,248 @@
+# Routing
+
+To serve incoming requests for a Next.js application deployed on Cloudflare Pages, a worker needs to be able to accurately point the request to the correct route. Routes in a Next.js app can be built as either be functions that need to be ran on the server, or static files that can just be served. In both cases, there are a series of steps that have to be taken to determine the correct destination.
+
+When building an application, the Vercel CLI takes a project and generates a configuration file that defines how the application should be served. This configuration file is useful in determining what happens at each stage of the routing process.
+
+To offer the best support for Next.js applications, the worker needs to try and replicate the routing process that Vercel uses.
+
+## Routing Phases
+
+In order to explain how routing works, it is important to first understand the underlying stages that the system goes through. These are taken from the Vercel build output config, and interpreted in a way that can be used by the worker.
+
+The routing process is broken down into a series of different phases.
+
+- `none` (no phase specified)
+- `filesystem`
+- `rewrite`
+- `resource`
+- `miss`
+- `error`
+- `hit`
+
+Each phase is processed in a somewhat sequential order, depending on the outcome of the previous phase. At different points, certain configuration options are applied, the final destination may be updated, or the final response object is modified.
+
+After checking the [source routes](https://vercel.com/docs/build-output-api/v3/configuration#source-route) for each phase, the system will check the build output to determine if the file path exists or not. If it can find a record for the path, it runs the `hit` phase and then breaks out of the routing system to return a response to the client.
```

Comment:
can you also mention that if there are errors it goes to `error`? (since here you are mentioning how the response is returned I think that it would be appropriate to mention the error scenario as well?)